numpy/linalg/linalg.py

```diff
@@ -2007,10 +2008,9 @@ def lstsq(a, b, rcond="warn"):
         b = b[:, newaxis]
     _assertRank2(a, b)
     _assertNoEmpty2d(a, b)  # TODO: relax this constraint
-    m  = a.shape[0]
-    n  = a.shape[1]
-    n_rhs = b.shape[1]
-    if m != b.shape[0]:
+    m, n = a.shape[-2:]
```

Comment:
Here, the assignments make the code a lot clearer (and of course now broadcastable); nice.